test/e2e/admin_portal.go

```diff
@@ -0,0 +1,198 @@
+package e2e
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo"                           //nolint
+	. "github.com/onsi/gomega"                           //nolint
+	. "github.com/serge1peshcoff/selenium-go-conditions" //nolint
+	. "github.com/tebeka/selenium"                       //nolint
+)
+
+var _ = Describe("Admin Portal E2E Testing", func() {
+	BeforeEach(skipIfNotInDevelopmentEnv)
+	var wdPoint *WebDriver
+	var wd WebDriver
+	var host string
+	JustBeforeEach(func() {
+		host, wdPoint = adminPortalSessionSetup()
+		wd = *wdPoint
+		wd.Get(host + "/v2")
+		wd.Refresh()
+	})
+
+	AfterEach(func() {
+		if wd != nil {
+			wd.Quit()
+		}
+	})
+	It("Should be able to populate cluster data correctly", func() {
+		err := wd.Wait(ElementIsLocated(ByCSSSelector, "div[data-automation-key='name']"))
+		if err != nil {
+			TakeScreenshot(wd, err)
+		}
```

Comment:
It is not evident from this code block, and all 20-odd that follow the same pattern, that the `TakeScreenshot` function is going to panic on purpose. IMO we should either change the function name or the way we handle these to be more explicit.

Simplest option is to change the name to something like `SaveScreenshotAndPanic`, but ideally we would change the behaviour to not panic and exit with grace. That's just my personal preference though :)

Reply:
Fixed